Bottom: 3c847cb0e15dc309a3b1c45c7462e0ea6ba96849
Top:    dee9667e6f721e6b7bc66e5408734235af925231
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-07 12:23:11 +0800

autofs-5.0.7 - add null check in read_one()

The pointter p shouldn't be null here but add a chack anyway.


---

diff --git a/modules/lookup_file.c b/modules/lookup_file.c
index 65e5ee6..2836996 100644
--- a/modules/lookup_file.c
+++ b/modules/lookup_file.c
@@ -302,8 +302,10 @@ static int read_one(unsigned logopt, FILE *f, char *key, unsigned int *k_len, ch
 				if (gotten == got_real || gotten == getting)
 					goto got_it;
 			} else if (mapent_len < MAPENT_MAX_LEN) {
-				mapent_len++;
-				*(p++) = ch;
+				if (p) {
+					mapent_len++;
+					*(p++) = ch;
+				}
 				nch = getc(f);
 				if (nch == EOF &&
 				   (gotten == got_real || gotten == getting))
